staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -324,9 +324,12 @@ func RESTClientFor(config *Config) (*RESTClient, error) {
 	var httpClient *http.Client
 	if transport != http.DefaultTransport {
 		httpClient = &http.Client{Transport: transport}
-		if config.Timeout > 0 {
-			httpClient.Timeout = config.Timeout
-		}
+		// do not set a timeout on the http client, instead use context for cancellation
+		// if multiple timeouts were set, the request will pick the smaller timeout to be applied, leaving other useless.
```

Comment:
> if multiple timeouts were set, the request will pick the smaller timeout to be applied, leaving other useless

if multiple timeouts were set, I would *expect* the smallest to rule

Reply:
if this is what we want for client-go then sure. As it is today sharing the same client for normal requests and informers and setting a `conig.Timeout` to `10 seconds` will terminate watches created by informers every `10 seconds`.

Reply:
You should not have to create two clients to use an informer.  client-go should let you specifically set a timeout for long running requests distinct from normal requests, and it should be clear what it does.